Recurly/BaseClient.cs

```diff
@@ -106,6 +107,11 @@ private RestRequest BuildRequest(Method method, string url, Request body = null,
             var request = new RestRequest(url, method);
             request.JsonSerializer = Recurly.JsonSerializer.Default;
 
+            if (options != null)
```

Comment:
Wondering if we shouldn't just initialize a default request options object instead then we don't need to do null checks everywhere. I could go either way.

```
if (options == null)
  options = new RequestOptions();
```

Reply:
Or if csharp supports it:

```
, RequestOptions options = new RequestOptions())
```

Reply:
I went with your original suggestion.

Setting a default value in the method declaration would have to be done in the definition of the operation method. Still the `MakeRequest(Async)` and `BuildRequest` methods declare `options` as a nullable optional parameter. The `options` parameter could be moved before the other optional parameters in those methods, but I'm not sure if it's worth it.